src/main/java/org/example/hw23/task4/Main.java

```diff
@@ -0,0 +1,16 @@
+package org.example.hw23.task4;
+
+import java.util.Optional;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
```

Comment:
If you're using BlockingQueue, there is no need to use low-level sync primitives such as wait/notify, sync blocks.
Just use BlockingQueue API to work with it -- take(), put(). They are blocking under the hood.

Or, if you already used this primitives, you can use just List/Queue, array. Anything except blocking queue